backend/src/test/asegroup1/api/models/landregistry/LandRegistryDataTest.java

```diff
@@ -0,0 +1,404 @@
+/**
+ * 
+ */
+package asegroup1.api.models.landregistry;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.security.InvalidParameterException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.EnumSet;
+import java.util.Random;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import asegroup1.api.models.landregistry.LandRegistryQueryConstraint.EstateType;
+import asegroup1.api.models.landregistry.LandRegistryQueryConstraint.PropertyType;
+import asegroup1.api.models.landregistry.LandRegistryQueryConstraint.TransactionCategory;
+import asegroup1.api.models.landregistry.LandRegistryQuerySelect.Selectable;
+
+/**
+ * @author Richousrick
+ *
+ */
+class LandRegistryDataTest {
+
+	private LandRegistryData lRData;
+
+	@BeforeEach
+	void initLandRegistryData() {
+		lRData = new LandRegistryData();
+	}
+
+	static String generateRandomString() {
+		return "Random: " + Math.random();
```

Comment:
### Error
All random values should be generated from a seed value this seed value can be randomly generated but needs to be output somewhere so that failed tests are reproducible.